client/src/pages/JoinPage/index.tsx

```diff
@@ -22,19 +26,74 @@ const JoinPage = ({
   const { stream, setStream } = useContext(DeviceContext);
   const { room_id } = useParams<{ room_id: string }>();
   const [showNotification, setShowNotification] = useState(false);
+  const [notificationText, setNotificationText] = useState("");
```

Comment:
What do you think about combining both showNotification and notificationText as something like notificationState, like
```javascript
const [notificationState, setNotificationState] = useState({show: false, text: ""})
```

Reply:
I agree yeah that seems better